include/sentry.h

```diff
@@ -834,12 +834,23 @@ SENTRY_API void sentry_options_set_symbolize_stacktraces(
 SENTRY_API int sentry_options_get_symbolize_stacktraces(
     const sentry_options_t *opts);
 
+/**
+ * Adds a new pattern for *in-app* classification.
+ *
+ * When on-device symbolication is enabled (see
+ * `sentry_options_set_symbolize_stacktraces`), this will flag stack frames as
+ * *in-app*, which match this pattern.
+ * Matching is done on both symbols names and object file names they belong to.
```

Comment:
"Pattern" here is underspecified. If you compare with existing SDKs, such as JavaScript [documented here](https://docs.sentry.io/platforms/python/configuration/options/#in-app-include), this is only a prefix match on the module.

@untitaker @mitsuhiko - did we never support functions for this? Being able to do a prefix match on functions sounds pretty essential, tbh.

Reply:
prefix match on the module was entirely sufficient for python. function names of a library don't share a common pattern one could match against

istm native just needs a different API, but that's fine IMO